d3ploy/no_inst.py

```diff
@@ -152,6 +152,19 @@ def tock(self):
                 for i in range(int(number)):
                     self.context.schedule_build(self, proto)
                     i += 1
+            if diff > 0:
+                proto = random.choice(self.prototypes)
+                if proto in self.fac_supply[commod]:
+                    prod_rate = self.fac_supply[commod][time]
```

Comment:
I think you mean `fac_supply[commod][proto]` here?

Reply:
yes. Then the loop can just looks for that type of proto.